### PR TITLE
[monaco][electron] Fix onigasm loading

### PR DESCRIPTION
**Pull request by marechal-p**
Onigasm is a WebAssembly binary file used to run Textmate coloring.  Before
this commit, the file was read using `fetch`, which doesn't support `file`
scheme uris.  Electron is running its frontend from the disk, so calls to
separate resources would also use the `file` scheme.

This commit replaces the implicit `fetch` call by requesting the file using
`XMLHttpRequest` directly.

Signed-off-by: Paul Maréchal <paul.marechal@ericsson.com>

---

The problem was happening under Electron, this PR fixes it.

### CODE REVIEW

**Comment by marechal-p** [comment]
Hmm, maybe I asked for too many people to review, but it would be good to have this issue fixed before the Theia release tomorrow ;)

File: packages/monaco/src/browser/textmate/monaco-textmate-frontend-bindings.ts [hunk 1]
```diff
@@ -25,8 +25,29 @@ import { LanguageGrammarDefinitionContribution } from './textmate-contribution';
...
+
+        request.onreadystatechange = function () {
+            if (this.readyState === XMLHttpRequest.DONE) {
+                if (this.status === 200) {
+                    resolve(this.response);
+                } else {
+                    reject(new Error('Could not fetch onigasm'));
+                }
+            }
+        };
```
Comment (elaihau) [inline packages/monaco/src/browser/textmate/monaco-textmate-frontend-bindings.ts]: would the code be easier to read if we use `xhr-request` (https://www.npmjs.com/package/xhr-request#example)? on the other hand, with `new XMLHttpRequest()` we don't need to add a new dependency :)

**Comment by marechal-p** [comment]
Fixes: https://github.com/theia-ide/theia/issues/2198

**Comment by svenefftinge** [comment]
Shouldn't this be solved here https://github.com/NeekSandhu/onigasm/blob/a9da30dacaf4346c71c79f8f98bbc0b4ecad3473/src/onigasmH.ts#L46

**Comment by svenefftinge** [comment]
To allow `file:`scheme in electron, see https://github.com/electron/electron/pull/3945/files

**Pull request merged**